OpenRA.Game/Network/UPnP.cs

```diff
@@ -10,61 +10,67 @@
 #endregion
 
 using System;
-using System.Diagnostics;
-using System.Net;
 using System.Threading;
 using System.Threading.Tasks;
 
-using Open.Nat;
+using Mono.Nat;
 
 namespace OpenRA.Network
 {
 	public enum UPnPStatus { Enabled, Disabled, NotSupported }
 
 	public class UPnP
 	{
-		static NatDevice natDevice;
-		static Mapping mapping;
-		static bool initialized;
-
-		public static IPAddress ExternalIP { get; private set; }
-		public static UPnPStatus Status
+		public UPnPStatus Status
 		{
 			get
 			{
-				return initialized ? natDevice != null ?
+				return NatUtility.IsSearching ? device != null ?
 					UPnPStatus.Enabled : UPnPStatus.NotSupported : UPnPStatus.Disabled;
 			}
 		}
 
-		public static async Task DiscoverNatDevices(int timeout)
+		Mapping mapping;
+		INatDevice device;
+
+		public UPnP()
 		{
-			initialized = true;
+			NatUtility.DeviceFound += DeviceFound;
+			NatUtility.StartDiscovery();
+			Game.RunAfterDelay(Game.Settings.Server.NatDiscoveryTimeout, () =>
+			{
+				NatUtility.StopDiscovery();
+				if (device == null)
+					Console.WriteLine("NAT discovery timed out.");
+			});
```

Comment:
https://github.com/alanmcgovern/Mono.Nat/blob/310d63d3de64c4668c71e36bbf0a39e0796fba82/Mono.Nat/Searcher.cs#L43 actually makes this obsolete I think.